node/assertion_error.ts

```diff
@@ -89,7 +89,7 @@ export function inspectValue(val: unknown): string {
   return inspect(
     val,
     {
-      compact: false,
+      compact: true,
```

Comment:
TODO

Reply:
I'm not sure what goes wrong here, @wafuwafu13 could you explain?

Reply:
@bartlomieju 
`compact` was not specified correctly here (should be `!!opts[0].compat`).
https://github.com/denoland/deno_std/blob/f3f9baaf96466a86b9c62223900763e85de61837/node/util.ts#L43

I reimplemented `inspect` in this PR so that `compact: false` could be configured correctly and the AssertionError occurred.

Reply:
Ah perfect! Thanks